src/websocket_socket_adapter.ts

```diff
@@ -0,0 +1,108 @@
+import {IContainer, IInstanceWrapper} from 'addict-ioc';
+import * as http from 'http';
+import {Logger} from 'loggerhythm';
+
+import {IHttpSocketAdapter} from '@essential-projects/http_socket_adapter_contracts';
+import {IIdentity, IIdentityService} from '@essential-projects/iam_contracts';
+import {EndpointSocketScope} from '@essential-projects/websocket';
+import {IEndpointSocketScope, OnConnectCallback} from '@essential-projects/websocket_contracts';
+
+import * as WebSocket from 'ws';
+
+const logger: Logger = Logger.createLogger('http:websocket_client');
+
+interface VerifyClientCallback {
+  (res: boolean, code?: number, message?: string, headers?: http.OutgoingHttpHeaders): void;
+}
+
+interface VerifyClientInfo {
+  origin: string;
+  secure: boolean;
+  req: http.IncomingMessage;
+}
+
+export class WebsocketHttpSocketAdapter implements IHttpSocketAdapter, IEndpointSocketScope {
```

Comment:
Why the mix? 
`IEndpointSocketScope` doesn't contain anything that isn't already located in `IHttpSocketAdapter`.

I also find it very misleading to implement an **adapter** based on a **scope** related interface.

```suggestion
export class WebsocketHttpSocketAdapter implements IHttpSocketAdapter {
```